applications/DEMApplication/python_scripts/DEM_analysis_stage.py

```diff
@@ -330,71 +331,98 @@ def GetProblemNameWithPath(self):
-        rigidFace_mp_filename = self.GetFemFilename()
-        if os.path.isfile(rigidFace_mp_filename+".mdpa"):
-            model_part_io_fem = self.model_part_reader(rigidFace_mp_filename, max_node_Id + 1, max_elem_Id + 1, max_cond_Id + 1)
-            model_part_io_fem.ReadModelPart(self.rigid_face_model_part)
-        else:
-            self.KratosPrintInfo('No mdpa file found for DEM walls. Continuing.')
-
-        max_node_Id = max(max_node_Id, self.creator_destructor.FindMaxNodeIdInModelPart(self.rigid_face_model_part))
-        max_elem_Id = max(max_elem_Id, self.creator_destructor.FindMaxElementIdInModelPart(self.rigid_face_model_part))
-        max_cond_Id = max(max_cond_Id, self.creator_destructor.FindMaxConditionIdInModelPart(self.rigid_face_model_part))
-        clusters_mp_filename = self.GetClusterFilename()
-        if os.path.isfile(clusters_mp_filename+".mdpa"):
-            model_part_io_clusters = self.model_part_reader(clusters_mp_filename, max_node_Id + 1, max_elem_Id + 1, max_cond_Id + 1)
-            model_part_io_clusters.ReadModelPart(self.cluster_model_part)
-        else:
-            self.KratosPrintInfo('No mdpa file found for DEM clusters. Continuing.')
-
-        max_elem_Id = self.creator_destructor.FindMaxElementIdInModelPart(self.spheres_model_part)
-        if max_elem_Id != old_max_elem_Id_spheres:
-            self.creator_destructor.RenumberElementIdsFromGivenValue(self.cluster_model_part, max_elem_Id)
-
-        max_node_Id = max(max_node_Id, self.creator_destructor.FindMaxNodeIdInModelPart(self.cluster_model_part))
-        max_elem_Id = max(max_elem_Id, self.creator_destructor.FindMaxElementIdInModelPart(self.cluster_model_part))
-        max_cond_Id = max(max_cond_Id, self.creator_destructor.FindMaxConditionIdInModelPart(self.cluster_model_part))
-        DEM_Inlet_filename = self.GetInletFilename()
-        if os.path.isfile(DEM_Inlet_filename+".mdpa"):
-            model_part_io_demInlet = self.model_part_reader(DEM_Inlet_filename, max_node_Id + 1, max_elem_Id + 1, max_cond_Id + 1)
-            model_part_io_demInlet.ReadModelPart(self.dem_inlet_model_part)
+    def ReadModelPartsFromMdpaFile(self, max_node_id, max_elem_id, max_cond_id):
+        def UpdateMaxIds(max_node_id, max_elem_id, max_cond_id, model_part):
+            max_node_id = max(max_node_id, self.creator_destructor.FindMaxNodeIdInModelPart(model_part))
+            max_elem_id = max(max_elem_id, self.creator_destructor.FindMaxElementIdInModelPart(model_part))
+            max_cond_id = max(max_cond_id, self.creator_destructor.FindMaxConditionIdInModelPart(model_part))
+            return max_node_id, max_elem_id, max_cond_id
+
+        def ReadModelPart(model_part, file_name, max_node_id, max_elem_id, max_cond_id):
+            file_path = self.GetInputFilePath(file_name)
+
+            if not os.path.isfile(file_path + '.mdpa'):
+                self.KratosPrintInfo('Input file ' + file_name + '.mdpa' + ' not found. Continuing.')
```

Comment:
Here `filename `only contains 'DEM' or 'DEMInlet' (the tag), it would be better if we print `file_path`. The name of the variable confused me quite a bit.

Reply:
idem as above